packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -1,241 +1,167 @@
+   */
   onDismiss?: () => void;
 }
 
-const DismissableLayerImpl = React.forwardRef<
-  DismissableLayerImplElement,
-  DismissableLayerImplProps
->((props, forwardedRef) => {
-  const {
-    disableOutsidePointerEvents = false,
-    onEscapeKeyDown,
-    onPointerDownOutside,
-    onFocusOutside,
-    onInteractOutside,
-    onDismiss,
-    ...layerProps
-  } = props;
-
-  const totalLayerCount = useTotalLayerCount();
-  const prevRunningLayerCount = usePreviousRunningLayerCount();
-  const runningLayerCount = prevRunningLayerCount + 1;
-  const isDeepestLayer = runningLayerCount === totalLayerCount;
-
-  const totalLayerCountWithDisabledOutsidePointerEvents =
-    useTotalLayerCountWithDisabledOutsidePointerEvents(disableOutsidePointerEvents);
-  const prevRunningLayerCountWithDisabledOutsidePointerEvents =
-    usePreviousRunningLayerCountWithDisabledOutsidePointerEvents();
-  const runningLayerCountWithDisabledOutsidePointerEvents =
-    prevRunningLayerCountWithDisabledOutsidePointerEvents + (disableOutsidePointerEvents ? 1 : 0);
-  const containsChildLayerWithDisabledOutsidePointerEvents =
-    runningLayerCountWithDisabledOutsidePointerEvents <
-    totalLayerCountWithDisabledOutsidePointerEvents;
-
-  // Disable pointer-events on `document.body` when at least one layer is disabling outside pointer events
-  useBodyPointerEvents({ disabled: disableOutsidePointerEvents });
-
-  // Dismiss on escape
-  useEscapeKeydown((event) => {
-    // Only dismiss if it's the deepest layer. his is effectively mimicking
-    // event.stopPropagation from the layer with disabled outside pointer events.
-    if (isDeepestLayer) {
-      onEscapeKeyDown?.(event);
-      if (!event.defaultPrevented) {
-        onDismiss?.();
-      }
-    }
-  });
-
-  // Dismiss on pointer down outside
-  const { onPointerDownCapture: handlePointerDownCapture } = usePointerDownOutside((event) => {
-    // Only dismiss if there's no deeper layer which disabled pointer events outside itself
-    if (!containsChildLayerWithDisabledOutsidePointerEvents) {
+const DismissableLayer = React.forwardRef<DismissableLayerElement, DismissableLayerProps>(
+  (props, forwardedRef) => {
+    const {
+      disableOutsidePointerEvents = false,
+      onEscapeKeyDown,
+      onPointerDownOutside,
+      onFocusOutside,
+      onInteractOutside,
+      onDismiss,
+      ...layerProps
+    } = props;
+    const context = React.useContext(DismissableLayerContext);
+    const [node, setNode] = React.useState<DismissableLayerElement | null>(null);
+    const [, force] = React.useState({});
+    const composedRefs = useComposedRefs(forwardedRef, (node) => setNode(node));
+    const [highestModal] = [...context.modals].slice(-1);
+    const highestModalIndex = [...context.layers].indexOf(highestModal);
+    const layerIndex = node ? [...context.layers].indexOf(node) : -1;
+    const hasPointerEvents = layerIndex >= highestModalIndex;
```

Comment:
`hasPointerEvents` confuses me and I keep having to double take it, because I don't know if it means `pointerEvents: 'auto'` or `pointerEvents: 'none'`.

Would renaming to `allowsPointerEvents` or `enablePointerEvents` clear that up?

Reply:
I read it like "this layer _hasPointerEvents_", so, pointer events are enabled. I've changed to `isPointerEventsEnabled` instead, if that helps?

Reply:
Yeah that works better for me, thanks!